CharlieBackend.Api.UnitTes/MentorServiceTests.cs

```diff
@@ -0,0 +1,202 @@
+        {
+            _accountServiceMock = new Mock<IAccountService>();
+            _notificationServiceMock = new Mock<INotificationService>();
+            _mapper = GetMapper(new ModelMappingProfile());
+        }
+
+        [Fact]
+        public async Task CreateMentorAsync()
+        {
+            //Arrange
+            int mentorExpectedId = 5;
+            var successExistingAccount = new Account()
+            {
+                Id = 1
+            };
+
+            var assignedExistingAccount = new Account()
+            {
+                Id = 2,
+                Role = UserRole.Mentor
+            };
+
+            _accountServiceMock.Setup(x => x.GetAccountCredentialsByIdAsync(1))
+                .ReturnsAsync(successExistingAccount);
+
+            _accountServiceMock.Setup(x => x.GetAccountCredentialsByIdAsync(2))
+                .ReturnsAsync(assignedExistingAccount);
+
+            var mentorRepositoryMock = new Mock<IMentorRepository>();
+            mentorRepositoryMock.Setup(x => x.Add(It.IsAny<Mentor>()))
+                .Callback<Mentor>(x => x.Id = mentorExpectedId);
+            _unitOfWorkMock.Setup(x => x.MentorRepository).Returns(mentorRepositoryMock.Object);
+
+            var mentorService = new MentorService(
+                _accountServiceMock.Object,
+                _unitOfWorkMock.Object,
+                _mapper,
+                _notificationServiceMock.Object);
+
+            //Act
+            var nonExistingIdResult = await mentorService.CreateMentorAsync(0);
+            var successResult = await mentorService.CreateMentorAsync(1);
+            var alreadyAssignedResult = await mentorService.CreateMentorAsync(2);
+
+            //Assert
+            Assert.Equal(ErrorCode.NotFound, nonExistingIdResult.Error.Code);
+
+            Assert.NotNull(successResult.Data);
+            Assert.Equal(successResult.Data.Id, mentorExpectedId);
+
+            Assert.Equal(ErrorCode.ValidationError, alreadyAssignedResult.Error.Code);
+        }
+
+        [Fact]
+        public async Task GetAllMentorsAsync()
+        {
+            //Arrange
+            var Mentors = new List<Mentor>() {
+                new Mentor() { },
+                new Mentor() { },
+                new Mentor() { }
+            };
+
+            var mentorRepositoryMock = new Mock<IMentorRepository>();
+            mentorRepositoryMock.Setup(x => x.GetAllAsync())
+                    .ReturnsAsync(Mentors);
+
+            _unitOfWorkMock.Setup(x => x.MentorRepository).Returns(mentorRepositoryMock.Object);
+
+            var mentorService = new MentorService(
+                _accountServiceMock.Object,
+                _unitOfWorkMock.Object,
+                _mapper,
+                _notificationServiceMock.Object);
+
+            //Act
+            var successResult = await mentorService.GetAllMentorsAsync();
+
+            //Assert
+            Assert.Equal(Mentors.Count, successResult.Count);
+        }
```

Comment:
You mock the return GetAll result of the repository, but the method GetAllMentorsAsync of the Mentor service has no other data processing logic, so if you mock repository GetAll method test is useless

Reply:
Yeah, agree with you, thought the same after PR